qiskit/backends/ibmq/api/ibmqconnector.py

```diff
@@ -0,0 +1,1097 @@
+        if user_id:
+            self.req.credential.set_user_id(user_id)
+        if not self.check_credentials():
+            raise CredentialsError('credentials invalid')
+        execution = self.req.get('/Executions/' + id_execution)
+        result = {}
+        if "result" in execution and "data" in execution["result"]:
+            if execution["result"]["data"].get('p', None):
+                result["measure"] = execution["result"]["data"]["p"]
+            if execution["result"]["data"].get('valsxyz', None):
+                result["bloch"] = execution["result"]["data"]["valsxyz"]
+            if "additionalData" in execution["result"]["data"]:
+                ad_aux = execution["result"]["data"]["additionalData"]
+                result["extraInfo"] = ad_aux
+            if "calibration" in execution:
+                result["calibration"] = execution["calibration"]
+            if execution["result"]["data"].get('cregLabels', None):
+                result["creg_labels"] = execution["result"]["data"]["cregLabels"]
+            if execution["result"]["data"].get('time', None):
+                result["time_taken"] = execution["result"]["data"]["time"]
+
+        return result
+
+    def get_code(self, id_code, access_token=None, user_id=None):
+        """
+        Get a code, by its id
+        """
+        if access_token:
+            self.req.credential.set_token(access_token)
+        if user_id:
+            self.req.credential.set_user_id(user_id)
+        if not self.check_credentials():
+            raise CredentialsError('credentials invalid')
+        code = self.req.get('/Codes/' + id_code)
+        executions = self.req.get('/Codes/' + id_code + '/executions',
+                                  '&filter={"limit":3}')
+        if isinstance(executions, list):
+            code["executions"] = executions
+        return code
+
+    def get_image_code(self, id_code, access_token=None, user_id=None):
+        """
+        Get the image of a code, by its id
+        """
+        if access_token:
+            self.req.credential.set_token(access_token)
+        if user_id:
+            self.req.credential.set_user_id(user_id)
+        if not self.check_credentials():
+            raise CredentialsError('credentials invalid')
+        return self.req.get('/Codes/' + id_code + '/export/png/url')
+
+    def get_last_codes(self, access_token=None, user_id=None):
+        """
+        Get the last codes of the user
+        """
+        if access_token:
+            self.req.credential.set_token(access_token)
+        if user_id:
+            self.req.credential.set_user_id(user_id)
+        if not self.check_credentials():
+            raise CredentialsError('credentials invalid')
+        last = '/users/' + self.req.credential.get_user_id() + '/codes/lastest'
+        return self.req.get(last, '&includeExecutions=true')['codes']
+
+    def run_experiment(self, qasm, backend='simulator', shots=1, name=None,
```

Comment:
This function seems to be the main function that kicks off an experiment. Shouldn't it take in a `Qobj` rather than qasm string? For devices not accepting qobj yet, it can read the qasm from inside the qobj.

Reply:
Actually this function is likely to be removed, as it is not invoked directly from terra - it was used by another consumer of `IBMQuantumExperience`.  For this PR, I think it is safer to start with something as close to the original package as possible, and tackle the cleaning up and other enhancements afterwards, including deciding on this function.